sql/src/main/java/org/apache/druid/sql/calcite/schema/SegmentMetadataCache.java

```diff
@@ -714,7 +715,12 @@ private Set<SegmentId> refreshSegmentsForDataSource(final String dataSource, fin
 
     log.debug("Refreshing metadata for dataSource[%s].", dataSource);
 
-    final long startTime = System.currentTimeMillis();
+    final ServiceMetricEvent.Builder builder =
+        new ServiceMetricEvent.Builder().setDimension(DruidMetrics.DATASOURCE, dataSource);
+
+    emitter.emit(builder.build("segment/metadatacache/refresh/count", segments.size()));
+
+    final long startNanos = System.nanoTime();
```

Comment:
Nit: This should probably be the first line of the method, so that we account for all the operations here.